In [ ]:
import pandas as pd
import numpy as np

In [ ]:
data=pd.read_csv('general_data.csv')
data

Age Attrition  ... YearsSinceLastPromotion YearsWithCurrManager
0      51        No  ...                       0                    0
1      31       Yes  ...                       1                    4
2      32        No  ...                       0                    3
3      38        No  ...                       7                    5
4      32        No  ...                       0                    4
...   ...       ...  ...                     ...                  ...
4405   42        No  ...                       0                    2
4406   29        No  ...                       0                    2
4407   25        No  ...                       1                    2
4408   42        No  ...                       7                    8
4409   40        No  ...                       3                    9

[4410 rows x 24 columns]

In [ ]:
gdata=data[data['Attrition']=='Yes']
gdata

Age Attrition  ... YearsSinceLastPromotion YearsWithCurrManager
1      31       Yes  ...                       1                    4
6      28       Yes  ...                       0                    0
13     47       Yes  ...                       9                    9
28     44       Yes  ...                       0                    0
30     26       Yes  ...                       0                    2
...   ...       ...  ...                     ...                  ...
4381   29       Yes  ...                       0                    1
4386   33       Yes  ...                       0                    4
4388   33       Yes  ...                       1                    7
4391   32       Yes  ...                       1                    2
4402   37       Yes  ...                       0                    0

[711 rows x 24 columns]

In [ ]:
print(gdata['DistanceFromHome'].mean())
print(gdata['DistanceFromHome'].std()) 

9.012658227848101
7.772367637890018


Statement 1 - The average distance from home is 9km with a standard deviation of 7.77. On taking a sample of 100 employees, the manager believes that if the distance from home of the employee is greater than 12km, attrition takes place.

In [ ]:
print(gdata['YearsSinceLastPromotion'].mean())
print(gdata['YearsSinceLastPromotion'].std())

1.9451476793248945
3.1486327816455715


Statement 2 - The average years of not getting a promotion after which an employee leaves the company is 1.9 years with a standard deviation of 3.15. The manager after taking a sample of 75 employees, came to a conclusion that if an employee gets a promotion within an year of joining, then attrition rate would decrease.

In [ ]:
print(gdata['TotalWorkingYears'].mean())
print(gdata['TotalWorkingYears'].std())

8.255289139633286
7.164017970571008


Satement  3 - The average Number of Years an employee work for the company is 8 with a standard deviation of 7.16. The manager believes that on taking sample of 100 employees attrition rate decreases if Number of working years is greater than 10.

In [ ]:
print(gdata['BusinessTravel'].value_counts())

Travel_Rarely        468
Travel_Frequently    207
Non-Travel            36
Name: BusinessTravel, dtype: int64


Statement 4 - The manager on taking a sample of 100 employees observed that employees who travelled frequently were less likely to leave their job when compared to employees who travelled rarely

In [ ]:
print(gdata['PercentSalaryHike'].mean())
print(gdata['PercentSalaryHike'].std())

15.481012658227849
3.775289126300602


Statement 5 - On taking a sampe of 100 employees, the manager observed that the mean percent salary hike was 15.48, with a standard deviation of 3.77. He believes that if the percent hike is less than 10%, attrition will take place.

In [ ]:
print(gdata['StockOptionLevel'].mean())
print(gdata['StockOptionLevel'].std())

0.7805907172995781
0.8588988626115427


Statement 6 - The average stock option level that the company offers the employees is 0.78, with a standard deviation of 0.86. The manager after taking a sample of 150 employees observes that attrition takes place if the stock options level is 1 or lesser.

In [ ]:
print(gdata['NumCompaniesWorked'].mean())
print(gdata['NumCompaniesWorked'].std())

2.9363507779349365
2.6787739217421955


Statement 7 - The average number of companies in which the employees have worked is 3, with a standard deviation of 2.7. The manager observed on taking a 100 employees that if the number of companies that the employee worked with is greater than 5, then attrition is more likely to happen.

In [ ]:
gdata['Education'].value_counts()

3    267
4    186
2    159
1     78
5     21
Name: Education, dtype: int64

Statement 8 - On observing a sample of 100 employees, the manager came to a conclusion that employees who had education of 3 were more likely to leave the company

In [ ]:
gdata['MaritalStatus'].value_counts()

Single      360
Married     252
Divorced     99
Name: MaritalStatus, dtype: int64

Statement 9 - The manager on observing a sample of 200 employees came to a conclusion that single people were more likely to undergo attrition when compared to married or divorced people.

In [ ]:
print(gdata['Age'].mean())
print(gdata['Age'].std())

33.607594936708864
9.675693299564738


Statement 10 - On taking a sample of 100 employees, it was obeserved that the average age of employees who were likely to leave the company was 33.6, with a standard deviation of 9.68. The manager observed that if the age was employees was less than 25, attrition was less likely to take place.